packages/frontend/src/hooks/cartesianChartConfig/useCartesianChartConfig.ts

```diff
@@ -18,13 +22,63 @@ type Args = {
     setPivotDimensions: React.Dispatch<
         React.SetStateAction<string[] | undefined>
     >;
+    columnOrder: string[];
+    explore: Explore | undefined;
+};
+
+export const sortDimensions = (
+    dimensionIds: string[],
+    explore: Explore | undefined,
+    columnOrder: string[],
+) => {
+    const getDimensionId = (dimension: Dimension) =>
+        `${dimension.table}_${dimension.name}`;
+
+    if (!explore) return dimensionIds;
+
+    const dimensions = getDimensions(explore);
+    const availableDimensionFields = dimensions.filter((dimension) =>
+        dimensionIds.includes(getDimensionId(dimension)),
+    );
+    const dateDimensions = availableDimensionFields.filter((dimension) =>
+        [DimensionType.DATE, DimensionType.TIMESTAMP].includes(dimension.type),
+    );
```

Comment:
We could avoid looping over the same data twice by combining the filters. 
filter if field is available and is time date/timestamp.